instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -646,9 +646,21 @@ def __init__(
         self.default_span_details = (
             default_span_details or get_default_span_details
         )
-        self.server_request_hook = server_request_hook
-        self.client_request_hook = client_request_hook
-        self.client_response_hook = client_response_hook
+
+        def failsafe(func):
+            @wraps(func)
+            def wrapper(span: Span, *args, **kwargs):
+                if func is not None:
```

Comment:
```suggestion
                if func is not None:
```

this should be checked outside `wrapper`